# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1366139117 14400
#      Tue Apr 16 15:05:17 2013 -0400
# Node ID 62d8aaf6ba8fa9c8299def12fe0c6a5cfb998116
# Parent  bfff338c56d15edc0ec02eea63e129f49a38e6ba
force update of GUI workspace viewer from clear function

* variables.cc (Fclear): Call octave_link::clear_workspace and
octave_link::set_workspace.

diff --git a/libinterp/interpfcn/variables.cc b/libinterp/interpfcn/variables.cc
--- a/libinterp/interpfcn/variables.cc
+++ b/libinterp/interpfcn/variables.cc
@@ -44,16 +44,17 @@ along with Octave; see the file COPYING.
 #include "defun.h"
 #include "dirfns.h"
 #include "error.h"
 #include "gripes.h"
 #include "help.h"
 #include "input.h"
 #include "lex.h"
 #include "load-path.h"
+#include "octave-link.h"
 #include "oct-map.h"
 #include "oct-obj.h"
 #include "ov.h"
 #include "ov-class.h"
 #include "ov-usr-fcn.h"
 #include "pager.h"
 #include "parse.h"
 #include "symtab.h"
@@ -2337,16 +2338,18 @@ without the dash as well.\n\
   string_vector argv = args.make_argv ("clear");
 
   if (! error_state)
     {
       if (argc == 1)
         {
           do_clear_globals (argv, argc, true);
           do_clear_variables (argv, argc, true);
+
+          octave_link::clear_workspace ();
         }
       else
         {
           int idx = 0;
 
           bool clear_all = false;
           bool clear_functions = false;
           bool clear_globals = false;
@@ -2448,16 +2451,20 @@ without the dash as well.\n\
                       symbol_table::clear_objects ();
                       octave_class::clear_exemplar_map ();
                     }
                   else
                     {
                       do_clear_symbols (argv, argc, idx, exclusive);
                     }
                 }
+
+              octave_link::set_workspace ((symbol_table::current_scope ()
+                                           == symbol_table::top_scope ()),
+                                          symbol_table::workspace_info ());
             }
         }
     }
 
   return retval;
 }
 
 DEFUN (whos_line_format, args, nargout,
